pyrealm/splash/evap.py

```diff
@@ -135,10 +147,10 @@ def estimate_aet(
         # subset the calculations to particular request days or use the entire array of
         # soil moisture. The slice here is used to programatically select `array[:]`.
         if day_idx is None:
-            check_input_shapes(wn, self.sat)
-            didx: int | slice = slice(self.sat.shape[0])
+            check_input_shapes(wn, shape=self.shape)
+            didx: int | slice = slice(self.shape[0])
```

Comment:
Ah. OK. This is the use case of the shape argument.  Hmmm...

I think the previous code was a bit of a lazy hack on my part - it's using `check_input_shapes` purely to raise an error. I think my comment on the suggested changes to `check_input_shapes` probably holds. I think it is cleaner here to do something like:

```suggestion
            if not wn.shape == self.shape:
                raise ValueError("The shape of wn does not match the existing SPLASH model data")
            didx: int | slice = slice(self.shape[0])
```

And then similar below. I think trying to dual purpose the function feels like it isn't keeping concerns separated?

Reply:
I agree about the separation of concerns regarding `check_input_shapes`, but I would be inclined to leave this from a practical point of view:

We can't compare `wn.shape` (or equivalents where this is used elsewhere) to `self.shape` directly because `wn` may not be fully broadcast, and we can't use `np.broadcast_shapes` because we require the same number of dimensions. So it would probably be best to define a new function to compare the shape.

But this function would necessarily be very similar to `check_input_shapes`, at which point I think it is simpler to instead use that function directly. What are your thoughts on this?

Reply:
@sjavis I see what you mean. And there is the additional rationale here that switching from `self.sat` to `self.shape` here is that `self.shape` isn't a single variable - it's the maximal size across all dimensions. And that is more robust - but it's a tuple of sizes, not an array so has to be passed in a different way.

OK - yup. Sold. Maybe in these use cases we wrap the `check_input_shapes` in a try block to provide more informative feedback in failure cases.

Reply:
Agreed, I will add some try blocks.

Reply:
I have added these in 7bdcdd0